draft-schwartz-add-ddr-forwarders.md

```diff
@@ -49,139 +49,124 @@ informative:
+# Conventions and Definitions
 
-This pattern has been deployed by Mozilla, with the domain "use-application-dns.net" {{MOZILLA-CANARY}}.
+Private IP Address - Any IP address reserved for loopback {{?RFC1122}}, link-local {{?RFC3927}}, private {{?RFC1918}}, local {{?RFC4193}}, or Carrier-Grade NAT {{?RFC6598}} use.
 
-### Service category restrictions
+Legacy DNS Forwarder - An apparent DNS resolver, known to the client only by a private IP address, that forwards the client's queries to an upstream resolver, and has not been updated with any knowledge of DDR.
 
-Certain DNS forwarders may block access to domains based on the category of service provided by those domains, e.g. domains hosting services that are not appropriate for a work or school environment.  As in the previous section, this requires an actively curated list of domains, because the set of domains that offer a given type of service is constantly changing.  An actively managed blocking list can easily be revised to include "resolver.arpa".
+Cross-Forwarder Upgrade - Establishment and use of a direct, encrypted connection between the client and the upstream resolver.
 
-### Time of use restrictions
+# Reputation Verified Selection (RVS) {#client-policy}
 
-Certain networks may impose restrictions on the time or duration of use by certain users.  This behavior is necessarily implemented below the DNS layer, because DNS-based blocking would be ineffective due to stub resolver caching, so it is not affected by changes in the DNS resolver.
+Reputation Verified Selection (RVS) is a method for validating whether connection using DDR is allowed.  Clients MAY use RVS when (a) the local DNS server is identified by a Private IP address and (b) the DDR SVCB resolution process does not produce any Encrypted DNS endpoints that have this IP address in their A or AAAA records.  RVS then proceeds as follows:
 
-## Upstream resolver services
+1. The client connects to one of the indicated Encrypted DNS endpoints.
+1. The client receives a certificate, which it verifies to a suitable root of trust.
+1. For each identity (e.g. SubjectAltName) in the certificate, the client constructs a Resolver Identity:
+  a. For DNS over TLS and DNS over QUIC, the Resolver Identity is an IP address or hostname and the port number used for the connection.
+  b. For DNS over HTTPS, the Resolver Identity is a URI Template in absolute form, containing the port number used for the connection and path indicated by `dohpath`.
+1. The client determines the reputation of each Resolver Identity derived from the certificate.
+1. The maximum (i.e. most favorable) reputation is the reputation of this connection.
 
-The forwarder's upstream resolver might provide additional services, such as filtering.  These services are generally independent of cross-forwarder upgrade, and hence naturally compatible.
+> TODO: Consider using the SVCB TargetName to select a single Resolver Identity.  Does this require the use of SNI?
```

Comment:
```suggestion
> OPEN QUESTION: Would it be better to use the SVCB TargetName to select a single Resolver Identity?  This would avoid the need to enumerate the certificate's names, but it would require the use of SNI (unlike standard DDR), and would not be compatible with all upstream encrypted resolvers.

> OPEN QUESTION: Can we simplify the resolver identity to just a domain name?  This would make reputation systems easier, but it would not allow distinct reputation for different colocated resolution services, so reputation providers would have to be sure that no approved resolver has other interesting colocated services.
```